src/exporter/DaemonMetricCollector.cc

```diff
@@ -33,7 +33,9 @@ void DaemonMetricCollector::request_loop(boost::asio::steady_timer &timer) {
   timer.async_wait([&](const boost::system::error_code &e) {
     std::cerr << e << std::endl;
     update_sockets();
-    dump_asok_metrics();
+    std::string dump_response;
+    std::string schema_response;
+    dump_asok_metrics(false, -1, true, dump_response, schema_response, true);
```

Comment:
This is pretty hacky and where it would be worth to move dump_response and schema_response generation.

Reply:
Where you suggest to move it to?

Reply:
dump_asok pseudo code looks something like:
```c
for (daemon in daemons) {
  ping
  get_dump_response
  get_schema_response
  get_config_show
  // do stuff
}
```

Then for each daemon iteration you have 4 variables that can be exchanged for that. Even with that in a testing phase all daemons will have same inputs which is not ok but whatever, works for this tests now.

I propose to change dump_asok_metrics a extrat the loop so that you have
```c
for (daemon in daemons) {
  ping
  get_dump_response
  get_schema_response
  get_config_show
  parse_daemon_metrics(&dump, &schema, &config);
}
```
so that in the test you can call `parse_daemon_metrics` that is what you care about, pinging is not a input this way, you remove unnecessary branches checking if this method is running in "test" mode or "normal" mode.

Reply:
Is good your idea @pereman2  - Even i think that probably "parse_daemon_metrics" must be a class for example, "generic_metrics_parser". 
As we have "special" parsing needs depending of the daemon, probably to have inherited classes for these daemons, for example "rgw_metrics_parser" could also simplify and make more consistent the code.

Reply:
@pereman2  @jmolmo I think this is addressed now given we have `parse_asok_metrics` introduced from  https://github.com/ceph/ceph/pull/55773 already & also I'm now simp,y using the config values for `sort_metrics` & `prio_limit` . PTAL  & lmk if anything more needs to be addressed.